src/desktop/lib/psi/PreloadLink.js

```diff
@@ -0,0 +1,134 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { Link, ResolverUtils, withRouter } from 'found'
+import { fetchQuery } from 'react-relay'
+import { isEmpty, isUndefined, omit } from 'lodash/fp'
+import { injectContext } from './AppProvider'
+
+const { getRouteMatches, getRouteValues } = ResolverUtils
+
+export const PreloadLink = withRouter(
+  injectContext(
+    context => {
+      return {
+        environment: context.environment,
+        routeConfig: context.routeConfig,
+        resolver: context.resolver,
+      }
+    },
+    class extends Component {
+      static propTypes = {
+        environment: PropTypes.object.isRequired,
+        immediate: PropTypes.bool, // load route data transparently in the bg
+        routeConfig: PropTypes.array.isRequired,
+        resolver: PropTypes.object,
+      }
+
+      static defaultProps = {
+        immediate: false,
```

Comment:
Just to be clear, does setting `immediate` to `false` mean data won't be fetched until the route is active?

Reply:
Yup! the idea is with the `relay-network-modern` we can actually get download progress, so eventually it will look something like this:

```javascript
<PreloadLink onProgress={progress => { ... }}>
  Hello
</PreloadLink>
```

Can also add `hover` and other subtle UI things so that the load feels snappy.